modin/experimental/xgboost/test/test_dmatrix.py

```diff
@@ -0,0 +1,166 @@
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+import pytest
+
+import modin.experimental.xgboost as mxgb
+import modin.pandas as pd
+import numpy as np
+
+import multiprocessing as mp
+import ray
+
+import xgboost as xgb
+
+from sklearn.metrics import accuracy_score
+
+
+ray.init(log_to_driver=False)
+
+num_cpus = mp.cpu_count()
+
+
+rng = np.random.RandomState(1994)
+
+
+def eval_general(data, modin_data, **kwargs):
+    try:
+        xgb.DMatrix(data, **kwargs)
+    except Exception:
+        with pytest.raises(Exception):
+            mxgb.DMatrix(modin_data, **kwargs)
+    else:
+        mxgb.DMatrix(modin_data, **kwargs)
+
+
+class TestDMatrix:
+    def test_dmatrix_numpy_init(self):
+        data = np.random.randn(5, 5)
+        dm = xgb.DMatrix(data)
+        md_dm = mxgb.DMatrix(pd.DataFrame(data))
+        assert md_dm.num_row() == dm.num_row() == 5
+        assert md_dm.num_col() == dm.num_col() == 5
+
+        data = np.array([[1, 2], [3, 4]])
+        dm = xgb.DMatrix(data)
+        md_dm = mxgb.DMatrix(pd.DataFrame(data))
+        assert md_dm.num_row() == dm.num_row() == 2
+        assert md_dm.num_col() == dm.num_col() == 2
+
+        for data in [
+            np.array([["a", "b"], ["c", "d"]]),
+        ]:
+            eval_general(data, pd.DataFrame(data))
+
+    def test_feature_names_slice(self):
+        data = np.random.randn(5, 5)
+
+        for f_n in [
+            list("abcdef"),
+            ["a", "b", "c", "d", "d"],
+            ["a", "b", "c", "d", "e<1"],
+        ]:
+            eval_general(data, pd.DataFrame(data), feature_names=f_n)
+
+        dm = xgb.DMatrix(data)
+        md_dm = mxgb.DMatrix(pd.DataFrame(data))
+        dm.feature_names = list("abcde")
+        md_dm.feature_names = list("abcde")
+        assert dm.feature_names == md_dm.feature_names == list("abcde")
+
+        dm = xgb.DMatrix(data, feature_types="q")
+        md_dm = mxgb.DMatrix(pd.DataFrame(data), feature_types="q")
+        assert dm.feature_types == md_dm.feature_types == list("qqqqq")
+
+        dm.feature_types = "q"
+        md_dm.feature_types = "q"
+        assert dm.feature_types == md_dm.feature_types == list("qqqqq")
+
+        dm.feature_types = list("qiqiq")
+        md_dm.feature_types = list("qiqiq")
+        assert dm.feature_types == md_dm.feature_types == list("qiqiq")
```

Comment:
We can expand your `eval_general` function to check equality of values of requested properties of DMatrix. It will help to reduce volume of code. Also, it would be better to move check of different values of properties to pytest params.